### PR TITLE
feature/user agent

### PR DESCRIPTION
**Pull request by baywet**
related https://github.com/microsoft/kiota/issues/2056

### CODE REVIEW

File: src/Middleware/UserAgentHandler.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            if (request.GetRequestOption<ObservabilityOptions>() is ObservabilityOptions obsOptions) {
+                activitySource = new ActivitySource(obsOptions.TracerInstrumentationName);
+                activity = activitySource?.StartActivity($"{nameof(UserAgentHandler)}_{nameof(SendAsync)}");
+                activity?.SetTag("com.microsoft.kiota.handler.useragent.enable", true);
+            } else {
+                activity = null;
+                activitySource = null;
+            }
+            try {
+                using var redirectActivity = activitySource?.StartActivity($"{nameof(UserAgentHandler)}_{nameof(SendAsync)}");
```
Comment (andrueastman) [inline src/Middleware/UserAgentHandler.cs:44]:
```suggestion
```
I believe this is unused and a duplicate of line 37

**Review by andrueastman — COMMENTED** [review]
Just a comment but looks good from my end.

File: src/Middleware/UserAgentHandler.cs (hunk 1 above)
Comment (baywet) [reply src/Middleware/UserAgentHandler.cs]: Copy pasta... 🤦‍♂️

**Review by andrueastman — APPROVED** [review]
👍🏼

**Pull request merged**